Doc/reference/expressions.rst

```diff
@@ -1745,6 +1745,11 @@ Or, when processing a file stream in chunks:
    while chunk := file.read(9000):
        process(chunk)
 
+Assignment expressions must be surrounded by parentheses when used
+as sub-expressions in subscript, slicing, conditional, lambda,
+keyword-argument, generator, and comprehension-if expressions.  They
```

Comment:
genexps don't need parens in 3.11:
```
>>> (x:=3 for y in [])
<generator object <genexpr> at 0x102ae97d0>
```
They completely disallow parens for the iterable (not sure why):
```
>>> (y for y in (z:=[]))
  File "<stdin>", line 1
SyntaxError: assignment expression cannot be used in a comprehension iterable expression
>>> (y for y in z:=[])
  File "<stdin>", line 1
    (y for y in z:=[])
                 ^^
SyntaxError: invalid syntax
```

Reply:
That's explained somewhere in [PEP 572](https://peps.python.org/pep-0572/#scope-of-the-target). "While it’s technically possible to assign consistent semantics to these cases, it’s difficult to determine whether those semantics actually make sense in the absence of real use cases."

Reply:
Oh interesting, I think it's actually this part a few paragraphs down: "Due to design constraints in the reference implementation (the symbol table analyser cannot easily detect when names are re-used between the leftmost comprehension iterable expression and the rest of the comprehension), named expressions are disallowed entirely as part of comprehension iterable expressions (the part after each “in”, and before any subsequent “if” or “for” keyword)"

Reply:
And so yet another wart in the specification was born... :-(